conans/test/functional/editable_reference/settings_and_options_test.py

```diff
@@ -0,0 +1,136 @@
+# coding=utf-8
+
+import os
+import unittest
+import tempfile
+from parameterized import parameterized
+
+from conans.test.utils.tools import TestClient
+from conans.paths.package_layouts.package_editable_layout import CONAN_PACKAGE_LAYOUT_FILE
+from conans.test import CONAN_TEST_FOLDER
+from conans.util.files import save
+from conans.paths import LINKED_FOLDER_SENTINEL
+from conans.model.ref import ConanFileReference
+
+
+class HeaderOnlyLibTestClient(TestClient):
+    header = """
+        #include <iostream>
+
+        void hello() {{
+            std::cout << "Hello {build_type}!" << std::endl;
+            std::cout << " - options.shared: {shared}" << std::endl;
+        }}
+        """
+
+    conanfile = """
+import os
+from conans import ConanFile, tools
+
+class Pkg(ConanFile):
+    name = "MyLib"
+    version = "0.1"
+
+    exports_sources = "*"
+    settings = "build_type"
+
+    options = {"shared": [True, False], }
+    default_options = {"shared": True, }
+
+    def package(self):
+        self.copy("*.hpp", dst="include", src="src/include")
+
+    def package_info(self):
+        self.cpp_info.libs = ["MyLib", "otra", ]
+        self.cpp_info.defines = ["MyLibDEFINES",]
+        self.cpp_info.libdirs = ["MyLib-libdirs", ]
+        self.cpp_info.includedirs = ["MyLib-includedirs", "dddd", ]
+
+    """
+
+    conan_package_layout = """
+[includedirs]
+src/include/{settings.build_type}/{options.shared}
+"""
+
+    def __init__(self, *args, **kwargs):
+        super(HeaderOnlyLibTestClient, self).__init__(*args, **kwargs)
+        self.save({"conanfile.py": self.conanfile,
+                   CONAN_PACKAGE_LAYOUT_FILE: self.conan_package_layout,
+                   "src/include/Debug/True/hello.hpp": self.header.format(build_type="Debug",
+                                                                          shared="True"),
+                   "src/include/Debug/False/hello.hpp": self.header.format(build_type="Debug",
+                                                                           shared="False"),
+                   "src/include/Release/True/hello.hpp": self.header.format(build_type="Release",
+                                                                            shared="True"),
+                   "src/include/Release/False/hello.hpp": self.header.format(build_type="Release",
+                                                                             shared="False"),
+                   })
+
+    def make_editable(self, full_reference):
+        conan_ref = ConanFileReference.loads(full_reference)
+        cache_dir = self.client_cache.conan(conan_ref)
+        save(os.path.join(cache_dir, LINKED_FOLDER_SENTINEL), content=self.current_folder)
```

Comment:
This test is testing the consuming part of the editable packages feature, it doesn't care about how that file is created there. (Also, it was created before having the command line argument).

Having the functionality separated into create -vs- consume in the tests is a good thing, IMHO, it will allow us to know if a future commit is breaking the creation of editable packages or the consumption (or both).